voluncheer/profiles/views/profile.py

```diff
@@ -97,26 +100,59 @@ def password_reset_request(request):
 
 def profile_update(request):
     """Get profile update POST and call save function on ChangeForms."""
-    profile = get_object_or_404(User, pk=request.user.pk)
     if request.user.is_volunteer:
+        profile = get_object_or_404(Volunteer, pk=request.user)
         form = VolunteerChangeForm(
             request.POST,
             request.FILES,
             instance=profile,
         )
     elif request.user.is_organization:
+        profile = get_object_or_404(Organization, pk=request.user)
         form = OrganizationChangeForm(request.POST, request.FILES, instance=profile)
     else:
         raise ValueError("profile_update: user must either a volunteer or an organizaiton.")
     form.save()
     return redirect("profile")
 
 
+# This part is for Volunteer specified features.
+
+
 def saved_events(request):
     volunteer = get_object_or_404(Volunteer, pk=request.user.pk)
     opportunity_selected = volunteer.opportunity_set.all()
     return render(
         request=request,
         template_name="profiles/savedevents.html",
-        context={"opportunity_selected": opportunity_selected},
+        context={
+            "opportunity_selected": opportunity_selected,
+            "volunteer": volunteer,
+            "opportunity_attended": [],
+        },
     )
+
+
+# This part is for Organization specified features.
+
+
+def confirm_attendance(request, opportunity_id):
+    """Get the request contains volunteer selection,
+    update opportunity's attended_volunteers attribute, and create archive object.
+    """
+    confirm_attendees = request.GET.getlist("volunteer-attended")
+    opportunity = get_object_or_404(Opportunity, pk=opportunity_id)
+    for attendee_pk in confirm_attendees:
+        volunteer = Volunteer.objects.get(pk=attendee_pk)
+        if volunteer not in opportunity.volunteers.all():
+            continue
+        if volunteer in opportunity.attended_volunteers.all():
+            opportunity.attended_volunteers.remove(volunteer)
+            volunteer.hours_volunteered -= opportunity.duration
```

Comment:
Why are we subtracting the hours here from the Volunteer? If they didn't attend, then they just don't get the hours added, right?

Reply:
This function will be triggered both for confirm attendance and cancel attendance, click again will cancel their attendance.

Reply:
every time the function is triggered, will it subtract the hours from a volunteer who didn't attend?

Reply:
Oh, I think I see now. The function checks to see if the volunteer is in attended_volunteers first, and if they are and we are canceling their attendance then they are removed?

Reply:
Yeah! That's correct!

Reply:
Got it! Just to confirm, in order to set a volunteer as not attended, the organization would have to click on the checkbox again?

Reply:
yeah, thats right